controllers/nstemplatetierrevisioncleanup/nstemplatetier_revision_cleanup_controller_test.go

```diff
@@ -0,0 +1,276 @@
+			}
+			//when
+			_, err := r.Reconcile(context.TODO(), req)
+			//then
+			require.EqualError(t, err, "unable to get the current NSTemplateTier: mock error")
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+
+		})
+		t.Run("error while listing outdated spaces", func(t *testing.T) {
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr, s, nsTemplateTier)
+			cl.MockList = func(ctx context.Context, list runtimeclient.ObjectList, opts ...runtimeclient.ListOption) error {
+				return fmt.Errorf("some error")
+			}
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			require.EqualError(t, err, "some error")
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+		})
+
+		t.Run("revision not found", func(t *testing.T) {
+			r, req, _ := prepareReconcile(t, "base")
+			//when
+			res, err := r.Reconcile(context.TODO(), req)
+			//then
+			require.NoError(t, err)
+			require.Equal(t, controllerruntime.Result{}, res)
+
+		})
+		t.Run("NSTemplate Tier not found", func(t *testing.T) {
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr, nsTemplateTier)
+			cl.MockGet = func(ctx context.Context, key types.NamespacedName, obj runtimeclient.Object, opts ...runtimeclient.GetOption) error {
+				if _, ok := obj.(*toolchainv1alpha1.NSTemplateTier); ok {
+					return errors.NewNotFound(schema.GroupResource{}, key.Name)
+				}
+				return cl.Client.Get(ctx, key, obj, opts...)
+			}
+			//when
+			res, err := r.Reconcile(context.TODO(), req)
+			//then
+			require.NoError(t, err)
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).DoNotExist()
+
+		})
+
+		t.Run("NSTemplate Tier not found, but error deleting ttr", func(t *testing.T) {
+			r, req, cl := prepareReconcile(t, ttr.Name, ttr, nsTemplateTier)
+			cl.MockGet = func(ctx context.Context, key types.NamespacedName, obj runtimeclient.Object, opts ...runtimeclient.GetOption) error {
+				if _, ok := obj.(*toolchainv1alpha1.NSTemplateTier); ok {
+					return errors.NewNotFound(schema.GroupResource{}, key.Name)
+				}
+				return cl.Client.Get(ctx, key, obj, opts...)
+			}
+			cl.MockDelete = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.DeleteOption) error {
+				return fmt.Errorf("some error cannot delete")
+			}
+			//when
+			res, err := r.Reconcile(context.TODO(), req)
+			//then
+			require.EqualError(t, err, "unable to delete the current Tier Template Revision base1ns-clusterresources-123456new-ttrcr: some error cannot delete")
+			require.Equal(t, controllerruntime.Result{}, res)
+			tiertemplaterevision.AssertThatTTRs(t, cl, nsTemplateTier.GetNamespace()).ExistFor(nsTemplateTier.Name)
+		})
+		t.Run("tier label not found", func(t *testing.T) {
+			ttr.Labels = map[string]string{}
```

Comment:
I wouldn't modify this object if it's being used by other tests, maybe we could create it's own ttr object